app/controllers/account/users_controller.rb

```diff
@@ -58,6 +58,18 @@ def destroy
     redirect_to account_users_path, flash: { success: 'User deleted!' }
   end
 
+  def generate_token
+    @user = resource
+    auth_token = @user.generate_auth_token
+    redirect_to account_user_path, flash: { success: "Your new API key is #{auth_token}" }
```

Comment:
never set workflow like this
send token on email or show token on the one time page so user could copy once